src/main/scala/com/typesafe/sbt/web/SbtWeb.scala

```diff
@@ -318,22 +304,50 @@ object SbtWeb extends AutoPlugin {
     nodeModules := nodeModuleGenerators(_.join).map(_.flatten).value
   )
 
+  def webJarsPathPrefix: Def.Initialize[Task[String]] = Def.task {
+    path(s"$WEBJARS_PATH_PREFIX/${moduleName.value}/${version.value}/")
+  }
+
+  def syncExportedAssets(track: TrackLevel): Def.Initialize[Task[File]] = Def.task {
+    val syncTargetDir = classDirectory.value
+    val syncRequired = TrackLevel.intersection(track, exportToInternal.value) match {
+      case TrackLevel.TrackAlways =>
+        true
+      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
+        !(syncTargetDir / webJarsPathPrefix.value).exists()
+    }
+    if (syncRequired) {
+      state.value.log.info(s"Exporting ${configuration.value}:${moduleName.value}")
+      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
+    } else
+      syncTargetDir
+  }
```

Comment:
@eed3si9n This is back to being a regular task. Thinking on it more, the `exportedMappings` are just mappings i.e. they are created as a result of scanning `webModuleDirectories`.

I think the gain from this PR so far is one where IO is avoided during the export of these assets.

If any of the sbt-web based compilers are being invoked each time an export is required, then I'm thinking that the problem could be with the way the compiler plugin has been written. For example, sbt-less uses sbt-js-engine which then also ensures that work isn't performed when not necessary via sbt-web's incremental task API (https://github.com/sbt/sbt-js-engine/blob/master/src/main/scala/com/typesafe/sbt/jse/SbtJsTask.scala#L267). You mentioned typescript elsewhere - It looks as though the sbt-typescript plugin has duplicated some sbt-js-engine code, so I'm unable to comment on whether it is using the incremental task API correctly.